homeassistant/components/http/__init__.py

```diff
@@ -240,6 +246,40 @@ def log(
         super().log(request, response, time)
 
 
+class HomeAssistantRequest(web.Request):
+    """Home Assistant request object."""
+
+    async def json(self, *, loads: JSONDecoder = json_loads) -> Any:
+        """Return body as JSON."""
+        # json_loads is a wrapper around orjson.loads that handles
+        # bytes and str. We can pass the bytes directly to json_loads.
+        return json_loads(await self.read())
+
+
+class HomeAssistantApplication(web.Application):
```

Comment:
I'm not 100% following what is going on here. `web.Application` is marked as `@final` and the `_make_request` function is part of `AppRunner`.

Reply:
Doesn't make any sense to me either after sleep.  I'll rework and re-run the traces. 

What I can't explain is I can see the new code being executed in the `py-spy`
<img width="1167" alt="Screenshot 2023-03-24 at 6 15 43 AM" src="https://user-images.githubusercontent.com/663432/227582123-01f178e8-f6ae-432d-8893-9d5d8ae50503.png">

Reply:
I definitely works as-is but I can't explain why .. yet

Reply:
After the aiohttp change for the logging no op merges (been through one review pass there already), I'll do another upstream PR to make this a bit easier for when 4.x is released